test/subdomainregistrar/TestForeverSubdomainRegistrar.js

```diff
@@ -0,0 +1,331 @@
+const { ethers } = require('hardhat')
+const { utils, BigNumber: BN } = ethers
+const { use, expect } = require('chai')
+const { solidity } = require('ethereum-waffle')
+const n = require('eth-ens-namehash')
+const namehash = n.hash
+const { FUSES } = require('../test-utils/ens')
+const { deploy } = require('../test-utils/contracts')
+
+const { CANNOT_UNWRAP, CAN_EXTEND_EXPIRY, PARENT_CANNOT_CONTROL } = FUSES
+
+use(solidity)
+
+const labelhash = (label) => utils.keccak256(utils.toUtf8Bytes(label))
+const ROOT_NODE =
+  '0x0000000000000000000000000000000000000000000000000000000000000000'
```

Comment:
ROOT_NODE and EMPTY_BYTES32 are the same so no need to hard code twice. It should rather use `const { EMPTY_BYTES32 } = require('../test-utils/constants')`